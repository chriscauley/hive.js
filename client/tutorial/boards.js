/* eslint-disable */
// not linting becauset his file should mostly be treated like binary

export default {
  queen: {"id":0.4963082061285773,"W":8,"H":6,"stacks":{"18":[2],"20":[4],"25":[7],"27":[5],"29":[1],"35":[3],"36":[6],"37":[0]},"piece_types":["queen","beetle","queen","ant","spider","ant","ant","grasshopper"],"piece_owners":[1,2,2,2,1,1,1,2],"hash":0.4918856798324476,"turn":20,"rules":{"piece_sets":["standard"],"no_rules":true}},
  beetle: {"id":0.9105569832241849,"W":6,"H":5,"stacks":{"8":[1],"13":[0,2],"14":[6],"15":[5],"20":[3],"21":[4]},"piece_types":["grasshopper","queen","beetle","queen","grasshopper","beetle","grasshopper"],"piece_owners":[2,2,1,1,1,2,2],"hash":0.6781573908095542,"turn":9,"rules":{"piece_sets":["standard"],"no_rules":true}},
  grasshopper: {"id":0.25727000168743475,"W":8,"H":4,"stacks":{"10":[2],"11":[0],"12":[1],"18":[4],"19":[3],"20":[5],"21":[6]},"piece_types":["queen","beetle","grasshopper","ant","queen","spider","grasshopper"],"piece_owners":[2,2,2,1,1,1,1],"hash":0.3070778715695497,"turn":9,"rules":{"piece_sets":["standard"],"no_rules":true}},
  spider: {"id":0.25727000168743475,"W":8,"H":6,"stacks":{"19":[0],"20":[1],"26":[5],"27":[3],"28":[4],"29":[2]},"piece_types":["spider","beetle","queen","queen","beetle","grasshopper"],"piece_owners":[1,1,1,2,2,2],"hash":0.8998216292260541,"turn":13,"rules":{"piece_sets":["standard"],"no_rules":true}},
  ant: {"id":0.8471811657173833,"W":8,"H":6,"stacks":{"12":[7],"18":[5],"20":[1],"25":[6],"27":[2],"28":[4],"29":[0],"35":[8],"37":[3]},"piece_types":["ant","spider","grasshopper","grasshopper","grasshopper","spider","queen","queen","ant"],"piece_owners":[2,1,1,2,2,1,1,2,1],"hash":0.03803776941801673,"turn":42,"rules":{"piece_sets":["standard"],"no_rules":true,"spiderwebs":true}},
  spider_webs: {"id":0.18315256412472647,"W":10,"H":6,"stacks":{"17":[4],"25":[2],"26":[3],"27":[5],"32":[7],"34":[1],"35":[0],"43":[6]},"piece_types":["spider","spider","ant","grasshopper","queen","ant","beetle","queen"],"piece_owners":[1,1,2,2,2,2,1,1],"hash":0.7745970078986681,"turn":10,"rules":{"piece_sets":["standard"],"no_rules":true, spiderwebs: true}},
  lady_bug: {"id":0.3931260941353747,"W":6,"H":5,"stacks":{"7":[0],"9":[5],"10":[8],"13":[1],"14":[2],"15":[4],"16":[6],"19":[7],"20":[3]},"piece_types":["lady_bug","grasshopper","beetle","queen","mosquito","queen","spider","ant","lady_bug"],"piece_owners":[2,1,1,1,2,2,1,2,1],"hash":0.7257064260146648,"turn":10,"rules":{"piece_sets":["standard","expanded_standard"],"no_rules":true}},
  mosquito: {"id":0.9055152935356559,"W":6,"H":6,"stacks":{"14":[0],"15":[3],"16":[5],"20":[1,6],"21":[2],"22":[4]},"piece_types":["scorpion","queen","grasshopper","mosquito","beetle","queen","mosquito"],"piece_owners":[1,1,1,2,1,2,1],"hash":0.16562214862327695,"turn":16,"rules":{"piece_sets":["standard","custom","expanded_custom","expanded_standard"],"no_rules":true}},
  pill_bug: {"id":0.8554086819938242,"W":6,"H":6,"stacks":{"8":[4],"14":[3],"15":[5],"19":[0],"20":[2],"25":[1]},"piece_types":["pill_bug","queen","spider","grasshopper","queen","ant"],"piece_owners":[1,1,1,2,2,2],"hash":0.14487339450473935,"turn":8,"rules":{"piece_sets":["standard","expanded_custom","custom","expanded_standard"],"no_rules":true}},
  mantis: {"id":0.23032420105806817,"W":8,"H":5,"stacks":{"11":[5,6],"12":[3],"18":[1],"20":[2],"21":[4],"27":[0]},"piece_types":["mantis","queen","beetle","queen","ant","ant","mantis"],"piece_owners":[1,1,2,2,2,1,2],"hash":0.37211475309241737,"turn":7,"rules":{"piece_sets":["standard","expanded_custom","custom","expanded_standard"],"no_rules":true}},
  fly: {"id":0.7696046353116126,"W":6,"H":5,"stacks":{"8":[6],"9":[7],"13":[5],"15":[4],"16":[3],"19":[0],"20":[1],"21":[2]},"piece_types":["fly","queen","beetle","grasshopper","fly","ant","lady_bug","queen"],"piece_owners":[1,1,1,1,2,2,2,2],"hash":0.37586779024891315,"turn":8,"rules":{"piece_sets":["standard","expanded_custom","custom","expanded_standard"],"spiderwebs":false,"no_rules":true}},
  scorpion: {"id":0.5599617752782116,"W":6,"H":5,"stacks":{"8":[2],"13":[1],"15":[3],"16":[5],"19":[0],"21":[4],"22":[6]},"piece_types":["queen","spider","beetle","queen","grasshopper","beetle","scorpion"],"piece_owners":[1,1,2,2,2,2,1],"hash":0.8561074820069865,"turn":11,"rules":{"piece_sets":["standard","expanded_custom","custom","expanded_standard"],"no_rules":true}},
  wasp: {"id":0.2979666972537396,"W":6,"H":6,"stacks":{"8":[1],"13":[0],"14":[2],"15":[4],"19":[3],"20":[5],"25":[6]},"piece_types":["ant","queen","wasp","queen","wasp","beetle","grasshopper"],"piece_owners":[2,2,2,1,1,1,2],"hash":0.7924051389594233,"turn":8,"rules":{"piece_sets":["standard","expanded_custom","custom","expanded_standard"],"spiderwebs":false,"no_rules":true}},
  dragonfly: {"id":0.4605378275656642,"W":6,"H":5,"stacks":{"7":[7],"8":[3],"14":[1,0],"15":[4],"19":[6],"20":[2],"21":[5]},"piece_types":["dragonfly","spider","queen","queen","lady_bug","beetle","dragonfly","spider"],"piece_owners":[1,1,1,2,2,2,2,1],"hash":0.7457731298416455,"turn":14,"rules":{"piece_sets":["standard","expanded_custom","custom","expanded_standard"],"no_rules":true}},
  centipede: {"id":0.26888780095322695,"W":6,"H":6,"stacks":{"10":[6],"14":[4],"15":[3,5],"20":[0],"21":[2],"27":[1]},"piece_types":["queen","grasshopper","centipede","ant","lady_bug","beetle","queen"],"piece_owners":[1,1,2,2,2,1,2],"hash":0.7771812831665879,"turn":10,"rules":{"piece_sets":["standard","expanded_custom","custom","expanded_standard"],"no_rules":true}},
  cockroach: {"id":0.49964340195479395,"W":8,"H":6,"stacks":{"12":[3],"14":[6],"18":[1],"19":[2],"20":[5],"21":[4,7],"27":[0]},"piece_types":["queen","cockroach","beetle","queen","fly","ant","cockroach","beetle"],"piece_owners":[1,1,2,2,2,1,2,1],"hash":0.5479472595173172,"turn":20,"rules":{"piece_sets":["standard","expanded_custom","custom","expanded_standard"],"no_rules":true}},
}