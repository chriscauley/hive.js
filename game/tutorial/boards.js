/* eslint-disable */
// not linting becauset his file should mostly be treated like binary

export default {
  queen: {"W":8,"H":6,"stacks":{"18":[2],"20":[4],"25":[7],"27":[5],"29":[1],"35":[3],"36":[6],"37":[0]},"piece_types":["queen","beetle","queen","ant","spider","ant","ant","grasshopper"],"piece_owners":[1,2,2,2,1,1,1,2],"turn":20,"rules":{}, "current_player": 1},
  beetle: {"W":6,"H":5,"stacks":{"8":[1],"13":[0,2],"14":[6],"15":[5],"20":[3],"21":[4]},"piece_types":["grasshopper","queen","beetle","queen","grasshopper","beetle","grasshopper"],"piece_owners":[2,2,1,1,1,2,2],"turn":9,"rules":{}, "current_player": 2},
  grasshopper: {"W":8,"H":4,"stacks":{"10":[2],"11":[0],"12":[1],"18":[4],"19":[3],"20":[5],"21":[6]},"piece_types":["queen","beetle","grasshopper","ant","queen","spider","grasshopper"],"piece_owners":[2,2,2,1,1,1,1],"turn":9,"rules":{}, "current_player": 2},
  spider: {"W":8,"H":6,"stacks":{"19":[0],"20":[1],"26":[5],"27":[3],"28":[4],"29":[2]},"piece_types":["spider","beetle","queen","queen","beetle","grasshopper"],"piece_owners":[1,1,1,2,2,2],"turn":13,"rules":{}, "current_player": 2},
  ant: {"W":8,"H":6,"stacks":{"12":[7],"18":[5],"20":[1],"25":[6],"27":[2],"28":[4],"29":[0],"35":[8],"37":[3]},"piece_types":["ant","spider","grasshopper","grasshopper","grasshopper","spider","queen","queen","ant"],"piece_owners":[2,1,1,2,2,1,1,2,1],"turn":42,"rules":{}, "current_player": 1},
  ladybug: {"W":6,"H":5,"stacks":{"7":[0],"9":[5],"10":[8],"13":[1],"14":[2],"15":[4],"16":[6],"19":[7],"20":[3]},"piece_types":["ladybug","grasshopper","beetle","queen","mosquito","queen","spider","ant","ladybug"],"piece_owners":[2,1,1,1,2,2,1,2,1],"turn":10,"rules":{}, "current_player": 1},
  mosquito: {"W":6,"H":6,"stacks":{"14":[0],"15":[3],"16":[5],"20":[1,6],"21":[2],"22":[4]},"piece_types":["scorpion","queen","grasshopper","mosquito","beetle","queen","mosquito"],"piece_owners":[2,1,1,2,1,2,1],"turn":16,"rules":{}, "current_player": 1},
  pill_bug: {"W":6,"H":6,"stacks":{"8":[4],"14":[3],"15":[5],"19":[0],"20":[2],"25":[1]},"piece_types":["pill_bug","queen","spider","grasshopper","queen","ant"],"piece_owners":[1,1,1,2,2,2],"turn":8,"rules":{},"current_player": 1},
  orchid_mantis: {"W":8,"H":6,"stacks":{"11":[5,6],"12":[3],"20":[2],"21":[4],"27":[0],"28":[1],"35":[7]},"piece_types":["orchid_mantis","queen","beetle","queen","ant","ant","orchid_mantis","grasshopper"],"piece_owners":[1,1,2,2,2,1,2,1],"turn":9,"rules":{}, "current_player": 2},
  fly: {"W":50,"H":50,"stacks":{"1176":[6],"1225":[5],"1227":[8],"1273":[4],"1274":[0,2],"1275":[1],"1276":[3],"1277":[7]},"piece_types":["queen","queen","fly","orchid_mantis","wasp","fly","wasp","scorpion","scorpion"],"piece_owners":[1,2,2,2,1,1,1,2,1],"turn":14,"rules":{},"current_player":1},
  scorpion: {"W":6,"H":5,"stacks":{"8":[2],"13":[1],"15":[3],"16":[5],"19":[0],"21":[4],"22":[6]},"piece_types":["queen","spider","beetle","queen","grasshopper","beetle","scorpion"],"piece_owners":[1,1,2,2,2,2,1],"turn":11,"rules":{}, "current_player": 2},
  wasp: {"W":6,"H":6,"stacks":{"8":[1],"13":[0],"14":[2],"15":[4],"19":[3],"20":[5],"25":[6]},"piece_types":["ant","queen","wasp","queen","wasp","beetle","grasshopper"],"piece_owners":[2,2,2,1,1,1,2],"turn":8,"rules":{}, "current_player": 1},
  dragonfly: {"W":6,"H":5,"stacks":{"7":[7],"8":[3],"14":[1,0],"15":[4],"19":[6],"20":[2],"21":[5]},"piece_types":["dragonfly","spider","queen","scorpion","queen","beetle","dragonfly","spider"],"piece_owners":[1,1,1,2,2,2,2,1],"turn":14,"rules":{}, "current_player": 1},
  damselfly: {"W":6,"H":5,"stacks":{"7":[7],"8":[3],"14":[1,0],"15":[4],"19":[6],"20":[2],"21":[5]},"piece_types":["damselfly","spider","queen","scorpion","queen","beetle","damselfly","spider"],"piece_owners":[1,1,1,2,2,2,2,1],"turn":14,"rules":{}, "current_player": 1},
  dragonfly_nymph: {"actions":[],"W":50,"H":50,"stacks":{"1225":[4],"1273":[2],"1274":[3,0],"1275":[1],"1323":[5]},"piece_types":["dragonfly_nymph","dragonfly_nymph","queen","queen","lanternfly_nymph","centipede"],"piece_owners":[1,2,1,2,2,1],"hash":"cb6ca89c583ff67546d37a5b4453bf1b714fe20e","turn":7,"rules":{},"last":{"to":1323},"current_player":2},
  centipede: {"W":6,"H":6,"stacks":{"10":[6],"14":[4],"15":[3,5],"20":[0],"21":[2],"27":[1]},"piece_types":["queen","grasshopper","centipede","ant","ladybug","beetle","queen"],"piece_owners":[1,1,2,2,2,1,2],"turn":10,"rules":{}, "current_player": 1},
  cockroach: {"W":8,"H":6,"stacks":{"12":[3],"14":[6],"18":[1],"19":[2],"20":[5],"21":[4,7],"27":[0]},"piece_types":["queen","cockroach","beetle","queen","fly","ant","cockroach","beetle"],"piece_owners":[1,1,2,2,2,1,2,1],"hash":0.5479472595173172,"turn":20,"rules":{}, "current_player": 1},
  cicada: {"actions":[],"W":50,"H":50,"stacks":{"1226":[3],"1273":[2],"1274":[0],"1275":[1],"1323":[4]},"piece_types":["queen","queen","cicada","cicada","ant"],"piece_owners":[1,2,1,2,1],"hash":"21f05bca4e8cc4257c4de80ac6dfdd6aaa971ec0","turn":5,"rules":{"pieces":{"beetle":2,"ant":3,"centipede":1,"cockroach":1,"cicada":2}},"current_player":2},
  lanternfly: {"W":50,"H":50,"stacks":{"1222":[6],"1225":[5],"1226":[7],"1273":[2],"1274":[0],"1275":[1],"1276":[3],"1323":[4]},"piece_types":["queen","beetle","beetle","queen","lanternfly","ant","cicada","lanternfly"],"piece_owners":[1,2,1,2,1,2,1,2],"hash":"c3ebd4ae845992bc447b61ee23449c23cfd8ea41","turn":8,"rules":{"pieces":{"beetle":2,"ant":3,"lanternfly":1,"centipede":1,"cockroach":1,"cicada":2}},"current_player":1},
  lanternfly_nymph: {"actions":[],"W":50,"H":50,"stacks":{"1223":[8],"1226":[5],"1273":[2],"1274":[0],"1275":[1],"1276":[3],"1323":[4],"1324":[6],"1326":[7]},"piece_types":["pill_bug","mosquito","queen","pill_bug","lanternfly_nymph","lanternfly_nymph","centipede","queen","orchid_mantis"],"piece_owners":[1,2,1,2,1,2,1,2,1],"hash":"3e89a125b257082f0a7cd6687cec8c0303eab5d3","turn":16,"rules":{},current_player: 1},
  trapdoor_spider: {"W":50,"H":50,"stacks":{"1226":[6],"1274":[0],"1275":[1],"1276":[3],"1277":[5],"1323":[4],"1325":[2],"1327":[7]},"piece_types":["trapdoor_spider","trapdoor_spider","queen","queen","cicada","ant","ant","cicada"],"piece_owners":[1,2,1,2,1,2,1,2],"rules": {},"current_player":2},
  orbweaver: {"actions":[],"W":50,"H":50,"stacks":{"1174":[10,8],"1176":[5],"1222":[12],"1225":[3],"1272":[6],"1273":[2],"1274":[0],"1275":[1],"1276":[7],"1325":[11],"1322":[9],"1323":[4]},"piece_types":["orbweaver","grasshopper","grasshopper","orbweaver","queen","queen","wasp","fly","fly","trapdoor_spider","trapdoor_spider","cockroach","cockroach"],"piece_owners":[1,2,1,2,1,2,1,2,1,2,1,2,1],"turn":18,"rules":{},"last":{"from":1226,"to":1277},"current_player":1,"last_move_at":1604858870793},
  praying_mantis: {"W":50,"H":50,"stacks":{"1225":[5],"1273":[6],"1274":[0,4],"1275":[1],"1276":[7,3],"1323":[2]},"piece_types":["queen","queen","praying_mantis","praying_mantis","beetle","fly","ant","beetle"],"piece_owners":[1,2,1,2,1,2,1,2],"turn":14,"last":{"from":1325,"special":1276},"current_player":1, rules: {}},
  emerald_wasp: {"W":50,"H":50,"stacks":{"1225":[5],"1273":[2],"1274":[0],"1275":[1,4],"1276":[3]},"piece_types":["beetle","beetle","queen","queen","emerald_wasp","emerald_wasp"],"piece_owners":[1,2,1,2,1,2],"turn":7,"rules":{},"current_player": 2},
  kung_fu_mantis: {"W":50,"H":50,"stacks":{"1224":[7],"1225":[4,3],"1274":[0],"1275":[1],"1276":[5],"1323":[6],"1324":[2],"1325":[8]},"piece_types":["queen","queen","ant","kung_fu_mantis","grasshopper","ant","kung_fu_mantis","ladybug","ladybug"],"piece_owners":[1,2,1,2,1,2,1,1,2],"turn":13,"rules":{},"current_player":2},
  earthworm: {"actions":[],"W":50,"H":50,"stacks":{"1223":[8],"1225":[3],"1226":[7],"1273":[0,6],"1274":[2],"1275":[1],"1276":[5],"1323":[4]},"piece_types":["centipede","pill_bug","queen","earthworm","earthworm","dragonfly","beetle","queen","lanternfly_nymph"],"piece_owners":[1,2,1,2,1,2,1,2,1],"hash":"1d4cc628d73f7cc48ab20a7b3a4800f37e13b3f3","turn":16,"rules":{},"current_player":1},
}