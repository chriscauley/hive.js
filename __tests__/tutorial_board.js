import boards from '../client/tutorial/boards'
import B from '../client/game/Board'
import { cloneDeep } from 'lodash'

test('Board snapshots', () => {
  const placements = {}
  const moves = {
    spiderwebs: '',
  }
  const specials = {}
  Object.entries(boards).forEach(([slug, board]) => {
    board = cloneDeep(board)
    B.update(board)
    const p1 = B.moves.getPlacement(board, 1)
    const p2 = B.moves.getPlacement(board, 2)
    placements[slug] = `${p1}|${p2}`
    moves[slug] = ''
    if (B.specials[slug]) {
      specials[slug] = ''
    }
    board.piece_types.forEach((type, piece_id) => {
      if (type !== slug) {
        return
      }
      board.rules = {}
      const s = piece_id + ':'
      moves[slug] += s + B.getMoves(board, piece_id).join(',') + '|'
      if (B.specials[slug]) {
        specials[slug] += s + B.getSpecials(board, piece_id).join(',') + '|'
      }
      if (slug === 'ant') {
        board.rules.spiderwebs = true
        moves.spiderwebs += s + B.getMoves(board, piece_id).join(',') + '|'
      }
    })
  })
  expect({ moves, specials, placements }).toMatchSnapshot()
})
