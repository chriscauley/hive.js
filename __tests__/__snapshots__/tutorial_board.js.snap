// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board snapshots 1`] = `
Object {
  "moves": Object {
    "ant": "0:21,13,4,11,19,10,17,16,24,30,38,45,36,43,34,26,33,24|8:26,33,24,16,17,10,19,11,4,36,45,38,30,22,21,13,4,11|",
    "beetle": "2:6,7,19,12,8,14|5:9,16,8,21,14|",
    "centipede": "2:16,22|",
    "cockroach": "1:28,35,26,29,13,22|6:|",
    "damselfly": "0:|6:13,25|",
    "dragonfly": "0:|6:|",
    "fly": "2:1224,1325,1324,1323,1222,1223,1272,1226,1174,1175,1327,1326,1126,1177|5:1176,1275|",
    "grasshopper": "2:4,29,26|6:2,27|",
    "lady_bug": "0:8,21,12,25,18|8:8,21,8,21,2,3,4,8,17,23,22,21,8,21|",
    "mantis": "0:|6:12,2,3,4,19,10|",
    "mosquito": "3:8,10,23,27,19,23|6:19,27,26,25,21|",
    "pill_bug": "0:13,18|",
    "queen": "0:30,45|2:|",
    "scorpion": "6:10,25|",
    "spider": "0:22,33,35|",
    "spider_webs": "",
    "spiderwebs": "0:21,13,4,11,19,30,38,45,36,43,34,26|8:26,33,24,16,17,10,19,11,4,36,45,38,30,22,21,13,4,11|",
    "super_grasshopper": "2:4,29,26,13,29,25,13,28,4|6:2,27,17,3,27,4,27,28,17,4,26,2|",
    "venom_centipede": "2:||S2:10,27|",
    "wasp": "2:10,16,21,27|4:7,2,9,6,31,24|",
  },
  "placements": Object {
    "ant": "10,16,17,24,26,33,34,43|4,11,13,22,30,38,45",
    "beetle": "6,12,22,25,26,27|2,9,10",
    "centipede": "25,26,28,33|4,11,13,17",
    "cockroach": "17,25,26,28,29,35|4,6,11,15,23",
    "dragonfly": "0,1,6,26|9,10,12,16,18,22",
    "fly": "1126,1174,1175,1177,1178,1222,1223,1272|1278,1324,1325,1326,1327",
    "grasshopper": "14,22,25,26,27,28,29|2,3,4,9",
    "lady_bug": "11,17,22,23,26,27|0,1,2,3,18",
    "mantis": "18,26,34,36,37,43|2,3,4,10,13,14,22",
    "mosquito": "13,19,25,26,27,28,29|9,10,17",
    "pill_bug": "12,18,24,26,27,31|2,7,9,10,16",
    "queen": "12,38,44,45|10,16,17,22,24,33,34",
    "scorpion": "6,12,18,25,28,29|2,9,10,17",
    "spider": "10,11,12,21,22,30|25,33,34,35,36",
    "spider_webs": "22,31,33,41,42,44,45,53|6,7,8,14,15,16,18,28,37",
    "wasp": "10,16,27|2,6,7,24,31",
  },
  "specials": Object {
    "centipede": "2:27|",
    "dragonfly": "0:7,9,16,12,27,25|6:26,24,17,21,6|",
    "mantis": "0:28,35|6:|",
    "mosquito": "3:|6:|",
    "pill_bug": "0:20,25|",
  },
}
`;
